ext/gitsh/src/line_editor.c

```diff
@@ -0,0 +1,1973 @@
+/************************************************
```

Comment:
ef19570 introduced an exact copy of the upstream module, and subsequent commits made changes to it.

You can see just the changes here: https://github.com/thoughtbot/gitsh/compare/ef19570...gb-custom-readline-extension#diff-9